### ISSUE TITLE
Improve the training procedure

### ISSUE THREAD

**Issue by vincekurtz** [vincekurtz/rddp#10]
The current training loop is fairly efficient, but lacks some features. Eventually we will want:

- [ ] A proper validation set (should be generated independently, since data points are correlated)
- [ ] Logging (to tensorboard or similar)
- [ ] Input normalization
